### DISCUSSION TITLE
docker containers stuck at STARTING

### DISCUSSION THREAD

**Discussion by jtgreen-cse** [Security-Onion-Solutions/securityonion#7845] [Unsupported Versions] [answered]
I'd like to describe a situation I got into and the fix, for posterity's sake.

I ran into an issue where a sensor node's `/nsm` filled up. After clearing some data, when the sensor came back up, `salt-minion` wasn't able to get the containers running again. I ran `salt-call -l all state.highstate` to see if it was blocking somewhere, and it sure was.

It was trying to get a list of the containers, but kept timing out (maybe, it was doing it over and over). I also noticed `so-status` was slow to show the container list, and all of them were showing `STARTING`. For fun I checked `docker ps -a` and wow, I had 100s of containers. I imagine calls to the docker socket were very slow and messed up anything trying to get details on docker's state. I ran `docker rm $(docker ps -qa)` to remove all containers, as soon as I did that, `state.highstate` unblocked, pulled everything from registry and brought everything back up.

**Comment by dougburks** [comment, marked as answer]
Thanks for sharing!
